libbeat/outputs/amqp/client.go

```diff
@@ -467,6 +472,8 @@ func (c *client) prepareEvent(codec codec.Codec, incoming eventTracker, now time
 	}
 	c.logger.Debugf("calculated routing key: %v", routingKey)
 
+	headers := c.getHeaders(content)
```

Comment:
Errors from inside `c.getHeaders` should be wrapped with contextual info and returned so that they can be handled here similar to other phases being executed, rather than internally logged and discarded. You should be able to see examples of this, such as in `encodeEvent`.